Shopify/cli
# Add version link to deploy banners

### WHY are these changes introduced?

Fixes https://github.com/Shopify/app-deploys/issues/517

IMPORTANT: This PR should only be merged once [this one](https://github.com/Shopify/partners/pull/48085) from partners is merged first 


### WHAT is this pull request doing?
- Retrieve new `location` field in the `appDeploy` operation
- Return the condition generated when the deployment is successful but not the release
- Add the version link to the body of the unified deploy successful messages
- Add a new info message when the unified deploy is successful but not the release

### How to test your changes?
- Create a new spin instance running `spin up partners -c partners.branch=app-deploy-add-location-field`
- Use an app with one ui extension with unified deployments rolled out
- Run the command `NODE_TLS_REJECT_UNAUTHORIZED=0 SHOPIFY_SERVICE_ENV=spin SPIN_INSTANCE=$(spin show -o name) pnpm shopify app deploy --path <YOUR_APP_PATH>` 

![image](https://github.com/Shopify/cli/assets/105213827/eec24cba-e0ce-4df6-abba-614dc6aedab6)

- Run the command `NODE_TLS_REJECT_UNAUTHORIZED=0 SHOPIFY_SERVICE_ENV=spin SPIN_INSTANCE=$(spin show -o name) pnpm shopify app deploy --path <YOUR_APP_PATH> --message "My new version"`  

![image](https://github.com/Shopify/cli/assets/105213827/40cdb1c3-5325-421c-9ddb-039902a44065)

- Run the command `NODE_TLS_REJECT_UNAUTHORIZED=0 SHOPIFY_SERVICE_ENV=spin SPIN_INSTANCE=$(spin show -o name) pnpm shopify app deploy --path <YOUR_APP_PATH> --message "My new version"  --no-release`  

![image](https://github.com/Shopify/cli/assets/105213827/3c3307c3-e40d-4b76-a6a4-8377c3d75fd1)

- I'm not able to reproduce the scenario that deploys successfully but fails releasing. There is a unit test for that. This is a capture of how the message should be displaying hacking the code to generate the error 😅

![image](https://github.com/Shopify/cli/assets/105213827/b3ab1b37-2726-4041-a889-ef842f1b9c02)

### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
